tobias/utils/motifs.py

```diff
@@ -733,20 +744,33 @@ def get_gimmemotif(self):
 	def get_biomotif(self):
 		""" Get biomotif object for motif """
 
+		# TODO implement
 		self.biomotif_obj = ""
 
 	def get_reverse(self):
 		""" Reverse complement motif """
-		if self.pfm is None:
-			self.get_pfm()
 
-		#Create reverse motif obj
-		reverse_motif = OneMotif()	#empty
-		for att in ["id", "name", "prefix", "strand", "length"]:
-			setattr(reverse_motif, att, getattr(self, att))
+		rev_strand = ["+" if s == "-" else "-" for s in self.strand]
+		# add 'rev_' prefix to indicate motif was reverse complemented
+		rev_id = "rev_" + self.id
+		rev_name = "rev_" + self.name if len(self.name) > 0  else self.name

```

Comment:
The name and id should stay the same - as it is still the same motif, just on the reverse strand :) In the end of a motif search, you mostly want to merge your results from "+" and "-" strand hits, and that is going to be hard without a common ID to merge on.

Reply:
Alright, consider this removed.